soh/src/overlays/actors/ovl_Door_Warp1/z_door_warp1.c

```diff
@@ -533,11 +533,15 @@ void DoorWarp1_ChildWarpOut(DoorWarp1* this, PlayState* play) {
                 if (GameInteractor_Should(VB_GIVE_ITEM_FROM_BLUE_WARP, true, ITEM_GORON_RUBY)) {
                     Item_Give(play, ITEM_GORON_RUBY);
                 }
-                play->nextEntranceIndex = ENTR_DEATH_MOUNTAIN_TRAIL_0;
-                gSaveContext.nextCutsceneIndex = 0xFFF1;
+                if (GameInteractor_Should(VB_BLUE_WARP_APPLY_ENTRANCE_AND_CUTSCENE, true, this)) {
```

Comment:
Upon second look at all the changes here in this file, I don't think we need any of them. You should be able to override the destination of the blue warp the same way we are for rando & cutscene skips with the `VB_PLAY_TRANSITION_CS`